datadog_checks_dev/tests/tooling/commands/validate/test_package.py

```diff
@@ -0,0 +1,50 @@
+# (C) Datadog, Inc. 2022-present
+# All rights reserved
+# Licensed under a 3-clause BSD style license (see LICENSE)
+import os
+import sys
+
+import pytest
+from click.testing import CliRunner
+
+from datadog_checks.dev import run_command
+
+
+def _build_pyproject_file(authors):
+    return f'''
+[project]
+name = "datadog-my-check"
+authors = {authors}
+[tool.hatch.version]
+path = "datadog_checks/my_check/__about__.py"
+'''
+
+
+@pytest.mark.parametrize(
+    'authors,expected_exit_code,expected_output',
+    [
+        ('[{ name = "Datadog", email = "packages@datadoghq.com" }]', 0, '1 valid'),
+        ('[{ name = "Datadog"}]', 0, '1 valid'),
+        ('[{ name = "Datadog", email = "invalid_email" }]', 1, 'Invalid email'),
+    ],
+)
+def test_validate_package_validates_emails(authors, expected_exit_code, expected_output):
+    runner = CliRunner()
+
+    with runner.isolated_filesystem():
+        os.mkdir('my_check')
+
+        with open('my_check/pyproject.toml', 'w') as f:
+            f.write(_build_pyproject_file(authors))
+
+        os.makedirs('my_check/datadog_checks/my_check')
+        with open('my_check/datadog_checks/my_check/__about__.py', 'w') as f:
+            f.write('__version__ = "1.0.0"')
+
+        result = run_command(
+            [sys.executable, '-m', 'datadog_checks.dev', '-x', 'validate', 'package', 'my_check'],
```

Comment:
Can't use the runner due to global state?

Reply:
I had actually forgotten about the fact that I was still using the `runner` (though only for the `isolated_filesystem` thing), from a previous iteration of this test. Technically I guess I could've used another way to achieve that effect of `cd`-ing to a tempdir, but this works.

I did manage to make this test work using the `runner`, by manipulating global state with a fixture that cleared the root dir and some `@lru_cache` caches before and after each test. What finally prevented me from using this approach was that importing anything under `datadog_checks.dev.tooling.commands` (unavoidable) broke a [totally unrelated test](https://github.com/DataDog/integrations-core/blob/4d971a2c66f1b015b423f278d73fdae33db47c47/datadog_checks_dev/tests/tooling/commands/test_agent.py#L7-L17) (details in [this ticket](https://datadoghq.atlassian.net/browse/AI-2446)).